src/multibody/joint/joint-dense.hpp

```diff
@@ -30,6 +30,7 @@ namespace se3
       NQ = _NQ, // pb
       NV = _NV
     };
+    typedef double Scalar;
```

Comment:
Scalar (if really needed) should not be declared in such a deep source file. 
Is Scalar already introduced otherwhere in pinocchio? If not, I would prefer to have a dedicated PR to introduce it everywhere (and we must discuss how to). If yes, it is very unfortunate.